aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -2,347 +2,120 @@
-    Create {
-        value: u128,
-    },
-    Delta {
-        speculative_start_value: SpeculativeStartValue,
-        delta: SignedU128,
-        history: DeltaHistory,
-    },
-}
-
-// Aggregator snapshot is immutable struct, once created - value is fixed.
-// If we want to provide mutability APIs in the future, it should be
-// copy-on-write - i.e. a new aggregator_id should be created for it.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub enum AggregatorSnapshotState {
-    // Created in this transaction, with explicit value
-    Create {
-        value: SnapshotValue,
-    },
-    // Created in this transaction, via snapshot(&aggregator)
-    Delta {
-        base_aggregator: DelayedFieldID,
-        delta: SignedU128,
-    },
-    // Created in this transaction, via string_concat(prefix, &snapshot, suffix)
-    Derived {
-        base_snapshot: DelayedFieldID,
-        formula: SnapshotToStringFormula,
-    },
-    // Accessed in this transaction, based on the ID
-    Reference {
-        // always expensive/aggregated read
-        speculative_value: SnapshotValue,
-    },
-}
-
-#[derive(Debug)]
-pub struct AggregatorSnapshot {
-    // The identifier used to identify the aggregator.
-    #[allow(dead_code)]
-    id: DelayedFieldID,
-
-    state: AggregatorSnapshotState,
-}
-
-impl AggregatorSnapshot {
-    pub fn into(self) -> AggregatorSnapshotState {
-        self.state
-    }
-}
-
-/// Internal aggregator data structure.
-#[derive(Debug)]
-pub struct Aggregator {
-    // The identifier used to identify the aggregator.
-    id: AggregatorVersionedID,
-    // Describes an upper bound of an aggregator. If value of the aggregator
-    // exceeds it, the aggregator overflows.
-    pub max_value: u128,
-    // Describes a state of an aggregator.
-    pub state: AggregatorState,
-}
+use move_binary_format::errors::PartialVMResult;
+use std::collections::{btree_map::Entry, BTreeMap};
 
 fn get_delayed_field_value_from_storage(
     id: &DelayedFieldID,
     resolver: &dyn DelayedFieldResolver,
-    mode: DelayedFieldReadMode,
 ) -> Result<DelayedFieldValue, PanicOr<DelayedFieldsSpeculativeError>> {
     // TODO transform unexpected errors into PanicError
     resolver
-        .get_delayed_field_value(id, mode)
+        .get_delayed_field_value(id)
         .map_err(|_err| PanicOr::Or(DelayedFieldsSpeculativeError::NotFound(*id)))
 }
 
-impl Aggregator {
-    #[cfg(test)]
-    pub fn get_history(&self) -> Option<&DeltaHistory> {
-        match &self.state {
-            AggregatorState::Create { .. } => None,
-            AggregatorState::Delta { history, .. } => Some(history),
-        }
-    }
+/// Stores all information about aggregators (how many have been created or
+/// removed), what are their states, etc. per single transaction).
+#[derive(Default)]
+pub struct DelayedFieldData {
+    // All aggregator instances that exist in the current transaction.
+    delayed_fields: BTreeMap<DelayedFieldID, DelayedChange<DelayedFieldID>>,
```

Comment:
you can see here:

```
#[derive(Default)]
pub struct AggregatorData {
    // All aggregators that were created in the current transaction, stored as ids.
    // Used to filter out aggregators that were created and destroyed in the
    // within a single transaction.
    new_aggregators: BTreeSet<AggregatorVersionedID>,
    // All aggregators that were destroyed in the current transaction, stored as ids.
    destroyed_aggregators: BTreeSet<StateKey>,
    // All aggregator instances that exist in the current transaction.
    aggregators: BTreeMap<AggregatorVersionedID, Aggregator>,
    // All aggregator snapshot instances that exist in the current transaction.
    aggregator_snapshots: BTreeMap<DelayedFieldID, AggregatorSnapshot>,
}
```

gets replaced with 

```
#[derive(Default)]
pub struct DelayedFieldData {
    // All aggregator instances that exist in the current transaction.
    delayed_fields: BTreeMap<DelayedFieldID, DelayedChange<DelayedFieldID>>,
}
```

Reply:
merging aggregators and aggregator_snapshots maps into single delayed_fields map. 

and so function in context.rs that creates VMChangeSet doesn't need to do any modifications any more.